pkg/kn/commands/service/service_create.go

```diff
@@ -147,6 +147,17 @@ func replaceService(client v1alpha1.KnClient, service *serving_v1alpha1_api.Serv
 		if err != nil {
 			return err
 		}
+
+		// Copy over any existing annotation
+		if len(existingService.Annotations) != 0 {
+			if service.Annotations == nil {
+				service.Annotations = map[string]string{}
+			}
+			for k, v := range existingService.Annotations {
```

Comment:
Please don't copy over all annotations unconditionally but only those which are supposed to be immutable. The purpose of `replaceService` is to replace everything which is possible.

And if you copy over annotations why not also labels ? what's the difference.

Just copy over `serving.CreatorAnnotation` and `serving.UpdatorAnnotation` which are validated in serving.CreatorAnnotation